### PR TITLE
11649_curved_buttons

### PR DESCRIPTION
**Pull request by dominicmarmionswrve**
https://swrvedev.jira.com/browse/SWRVE-11649

@Sergio-Mira @adam-govan

### CODE REVIEW

File: SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/engine/model/ButtonControl.java [hunk 1]
```diff
@@ -2,8 +2,13 @@
 
 public class ButtonControl extends ControlBase {
     protected String description;
+    protected String border_radius;
 
     public String getDescription() {
         return description;
     }
+
+    public String getBorder_radius() {
```
Comment (Sergio-Mira) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/engine/model/ButtonControl.java]: getBorderRadius (will also work afaik, other atoms do this)

File: SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationButton.java [hunk 2]
```diff
@@ -37,6 +39,13 @@ public ConversationButton(Context context, ButtonControl model, int defStyle) {
         setEllipsize(TextUtils.TruncateAt.END);
     }
 
+    private void initBorderRadius() {
+        if(model.getBorder_radius() != null) {
+            borderRadius = Float.parseFloat(model.getBorder_radius());
+            // TODO convert from percent to correct units SWRVE-11668
```
Comment (Sergio-Mira) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationButton.java:45]: Unfinished?

File: SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationButton.java [hunk 3]
```diff
@@ -97,10 +103,21 @@ private Drawable getOutlineDrawable(int borderColor, int backgroundColor) {
         return new LayerDrawable(drawables);
     }
 
+    private ShapeDrawable getSolidDrawable(int color) {
+        // Each corner has 2 radii (xradius, yradius), giving 8 values
```
Comment (Sergio-Mira) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationButton.java:107]: :+1:

File: SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationButton.java (hunk 2 above)
Comment (dominicmarmionswrve) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/ui/ConversationButton.java:45]: Yeah. Hence the comment. Will remove as part of SWRVE-11668

File: SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/engine/model/ButtonControl.java (hunk 1 above)
Comment (dominicmarmionswrve) [inline SwrveConversationSDK/src/main/java/com/swrve/sdk/conversations/engine/model/ButtonControl.java]: :+1:

**Pull request merged**